modules/core/shared/src/main/scala/io/circe/JsonNumber.scala

```diff
@@ -264,20 +273,20 @@ final object JsonNumber {
     if (result.eq(null)) None else Some(JsonBiggerDecimal(result))
   }
 
+  private[this] val bigDecimalMinLong: JavaBigDecimal = new JavaBigDecimal(Long.MinValue)
+  private[this] val bigDecimalMaxLong: JavaBigDecimal = new JavaBigDecimal(Long.MaxValue)
+
+  private[circe] def bigDecimalIsWhole(value: JavaBigDecimal): Boolean =
+    value.signum == 0 || value.scale <= 0 || value.stripTrailingZeros.scale <= 0
```

Comment:
Yeah, I had problems with that and other Java <8 `stripTrailingZeros` bugs back when I was first putting together `BiggerDecimal`, but I think this should always be fine (because of the `signum` check and since the scale won't be close to `Int.MaxValue`).